Adding Zodiark Split keyboard

Aleblazer opened the pull request:
<!--- Provide a general summary of your changes in the title above. -->

<!--- This template is entirely optional and can be removed, but is here to help both you and us. -->
<!--- Anything on lines wrapped in comments like these will not show up in the final text. -->

## Description

Adding Aleblazer/Zodiark split keyboard and keymaps

## Types of Changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply. -->
- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

zvecr commented on keyboards/aleblazer/zodiark/zodiark.h:
```diff
@@ -0,0 +1,39 @@
+#include "split_util.h"
+
+#define LAYOUT( \
+ L00, L01, L02, L03, L04, L05, L06, R00, R01, R02, R03, R04, R05, R06, \
```
Top part of layout macro should represent physical layout, currently it does not match the image provided in the readme

zvecr commented on keyboards/aleblazer/zodiark/rules.mk:
```diff
@@ -0,0 +1,63 @@
+#     This will define a symbol, F_USB, in all source code files equal to the
+#     input clock frequency (before any prescaling is performed) in Hz. This value may
+#     differ from F_CPU if prescaling is used on the latter, and is required as the
+#     raw input clock is fed directly to the PLL sections of the AVR for high speed
+#     clock generation for the USB and other AVR subsections. Do NOT tack on a 'UL'
+#     at the end, this will be done automatically to create a 32-bit value in your
+#     source code.
+#
+#     If no clock division is performed on the input clock inside the AVR (via the
+#     CPU clock adjust registers or the clock division fuses), this will be equal to F_CPU.
+F_USB = $(F_CPU)
+
+# Interrupt driven control endpoint task(+60)
+OPT_DEFS += -DINTERRUPT_CONTROL_ENDPOINT
+
+
+# Boot Section Size in *bytes*
+OPT_DEFS += -DBOOTLOADER_SIZE=4096
+EXTRAFLAGS += -flto
+
+
+# Build Options
+#   comment out to disable the options.
+#
+BOOTMAGIC_ENABLE ?= no	# Virtual DIP switch configuration(+1000)
+MOUSEKEY_ENABLE ?= no	# Mouse keys(+4700)
+EXTRAKEY_ENABLE ?= yes	# Audio control and System control(+450)
+CONSOLE_ENABLE ?= no	# Console for debug(+400)
+COMMAND_ENABLE ?= no    # Commands for debug and configuration
+SLEEP_LED_ENABLE ?= no  # Breathing sleep LED during USB suspend
+NKRO_ENABLE ?= no		# USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+BACKLIGHT_ENABLE ?= no  # Enable keyboard backlight functionality
+AUDIO_ENABLE ?= no
+OLED_DRIVER_ENABLE = yes    # Enable Support for SSD1306 or SH1106 OLED Displays; Communicating over I2C
+ENCODER_ENABLE = yes
+SPLIT_KEYBOARD = yes
+LTO_ENABLE = yes
+RGBLIGHT_ENABLE = no
+RGB_MATRIX_ENABLE = no
+RGB_MATRIX_DRIVER = WS2812
```
```suggestion
# MCU name
MCU = atmega32u4

# Bootloader selection
BOOTLOADER = caterina

# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = no            # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
OLED_DRIVER_ENABLE = yes    # Enable Support for SSD1306 or SH1106 OLED Displays; Communicating over I2C
ENCODER_ENABLE = yes
SPLIT_KEYBOARD = yes
LTO_ENABLE = yes
RGBLIGHT_ENABLE = no
RGB_MATRIX_ENABLE = no
RGB_MATRIX_DRIVER = WS2812
```

zvecr commented on keyboards/aleblazer/zodiark/config.h:
```diff
@@ -0,0 +1,86 @@
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#ifndef CONFIG_H
+#define CONFIG_H
```
```suggestion
#pragma once
```

zvecr commented on keyboards/aleblazer/zodiark/config.h:
```diff
@@ -0,0 +1,86 @@
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    Aleblazer
+#define PRODUCT         Zodiark
+#define DESCRIPTION     68 key split keyboard
+
```
```suggestion

```

zvecr commented on keyboards/aleblazer/zodiark/config.h:
```diff
@@ -0,0 +1,86 @@
+#define RGB_MATRIX_VAL_STEP 8
+#define RGB_MATRIX_SPD_STEP 10
+#endif
+
+#endif
```
```suggestion
```

Aleblazer commented:
Applied all change requests, and thanks to ridingqwerty in the QMK Discord, fixed the zodiark.h layout and all the keymaps to reflect the shift in key locations.

fauxpark commented on keyboards/aleblazer/zodiark/readme.md:
````diff
@@ -0,0 +1,27 @@
+
+A split keyboard with 5x7 including a thumbcluster, encoders on each side, per key RGB, and 2x I2C headers per side, supporiting 1.3"/.96" 128x64 OLEDs (the 1.3" is an SSH1106 OLED, refer to QMK documentation for limitations), .91" 128x32 OLEDs.
+
+Keyboard Maintainer: [Aleblazer](https://github.com/Aleblazer/), [Discord Link](https://discord.gg/BCSbXwskVt)
+Hardware Supported: Pro Micro and derivatives
+Hardware Availability: Pending Group Buy - [Discord Link](https://discord.gg/BCSbXwskVt)
+
+Make example for this keyboard (after setting up your build environment):
+
+```sh
+make zodiark:default
+```
+
+Flash the default keymap:
+
+```sh
+make zodiark:default:avrdude
+```
````
```suggestion
* Keyboard Maintainer: [Aleblazer](https://github.com/Aleblazer/), [Discord Link](https://discord.gg/BCSbXwskVt)
* Hardware Supported: Pro Micro and derivatives
* Hardware Availability: Pending Group Buy - [Discord Link](https://discord.gg/BCSbXwskVt)

Make example for this keyboard (after setting up your build environment):

    make zodiark:default

Flashing example for this keyboard:

    make zodiark:default:flash

fauxpark commented on keyboards/aleblazer/zodiark/zodiark.h:
```diff
@@ -0,0 +1,39 @@
+*/
+#pragma once
+#include "quantum.h"
+#include "split_util.h"
```
```suggestion
```

fauxpark commented on keyboards/aleblazer/zodiark/rules.mk:
```diff
@@ -0,0 +1,28 @@
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
+AUDIO_ENABLE = no           # Audio output
+OLED_DRIVER_ENABLE = yes    # Enable Support for SSD1306 or SH1106 OLED Displays; Communicating over I2C
+ENCODER_ENABLE = yes
+SPLIT_KEYBOARD = yes
+LTO_ENABLE = yes
+RGBLIGHT_ENABLE = no
+RGB_MATRIX_ENABLE = no
```
```suggestion
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
OLED_DRIVER_ENABLE = yes    # Enable Support for SSD1306 or SH1106 OLED Displays; Communicating over I2C
ENCODER_ENABLE = yes
SPLIT_KEYBOARD = yes
LTO_ENABLE = yes
```

Aleblazer commented:
Applied all suggestions from fauxpark, and added RGB_MATRIX_DRIVER = WS2812 to my rgbmatrixwip keymap rules.mk.

drashna commented on keyboards/aleblazer/zodiark/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,314 @@
+#include QMK_KEYBOARD_H
```
Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

drashna commented on keyboards/aleblazer/zodiark/config.h:
```diff
@@ -0,0 +1,81 @@
+#define PRODUCT         Zodiark
+#define MASTER_LEFT
+
+#ifdef OLED_DRIVER_ENABLE
+  #define OLED_TIMEOUT 400000
+#endif
```
Is this is defined at the keyboard level, it probably doesn't need to be fined here.
```suggestion
```

drashna commented on keyboards/aleblazer/zodiark/keymaps/rgbmatrixwip/keymap.c:
```diff
@@ -0,0 +1,314 @@
+#include QMK_KEYBOARD_H
```
Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

drashna commented on keyboards/aleblazer/zodiark/keymaps/rgbmatrixwip/rules.mk:
```diff
@@ -0,0 +1,2 @@
+RGB_MATRIX_ENABLE = yes
+RGB_MATRIX_DRIVER = WS2812
```
This should be moved to the keyboard's config.

drashna commented on keyboards/aleblazer/zodiark/keymaps/slimoled/keymap.c:
```diff
@@ -0,0 +1,314 @@
+#include QMK_KEYBOARD_H
```
Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

Aleblazer commented:
Cleaned up the OLED code from the keyboard level config.h, added the GPL2+ license header to all keymaps, and removed the rgbmatrixwip keymap (Matrix split effects didn't work without a custom solution at this time).

drashna commented on keyboards/aleblazer/zodiark/config.h:
```diff
@@ -0,0 +1,77 @@
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    Aleblazer
+#define PRODUCT         Zodiark
+#define MASTER_LEFT
```
This isn't actually needed, as this is the default behavior. 
```suggestion
```

drashna commented on keyboards/aleblazer/zodiark/config.h:
```diff
@@ -0,0 +1,77 @@
+
+#define NO_ACTION_MACRO
+
+#define USE_SERIAL
```
This is the default behavior, so this isn't needed.
```suggestion
```

fauxpark commented on keyboards/aleblazer/zodiark/readme.md:
```diff
@@ -0,0 +1,23 @@
+# Zodiark Keyboard
+
+![Zodiark Split](https://i.imgur.com/49O8aow.jpg)
```
```suggestion
![Zodiark Split](https://i.imgur.com/49O8aowl.jpg)
```

Aleblazer commented:
All requested changes have been made, and added a VIA keymap.

zvecr commented:
Thanks"
